src/almanack/git.py

```diff
@@ -183,40 +186,51 @@ def find_file(
             The path to the file within the repository.
         case_insensitive (bool):
             If True, perform case-insensitive comparison.
+        extensions (list[str]):
+            List of possible file extensions to check (e.g., [".md", ""]).
 
     Returns:
         Optional[pygit2.Object]:
             The entry of the found file,
             or None if no matching file is found.
     """
+    # Get the tree object of the latest commit
     tree = repo.head.peel().tree
-    found_entry = None
 
-    if not case_insensitive:
-        try:
-            found_entry = tree[filepath]
-        except KeyError:
-            return None
-    else:
-        path_parts = filepath.lower().split("/")
-        for i, part in enumerate(path_parts):
+    # Iterate over each extension to check for the file
+    for ext in extensions:
+        full_path = f"{filepath}{ext}"  # Construct the full path with the extension
+        if not case_insensitive:
             try:
-                entry = next(e for e in tree if e.name.lower() == part)
-            except StopIteration:
-                return None
-
-            if entry.type == pygit2.GIT_OBJECT_TREE:
-                tree = repo[entry.id]
-            elif entry.type == pygit2.GIT_OBJECT_BLOB:
-                if i == len(path_parts) - 1:
-                    found_entry = entry
-                    break
+                # Try to get the file entry directly (case-sensitive)
+                return tree[full_path]
+            except KeyError:
+                continue  # If not found, continue to the next extension
+        else:
+            # Split the path into parts for case-insensitive comparison
+            path_parts = full_path.lower().split("/")
+            current_tree = tree
+            for i, part in enumerate(path_parts):
+                try:
+                    # Find the entry in the current tree that matches the part (case-insensitive)
+                    entry = next(e for e in current_tree if e.name.lower() == part)
+                except StopIteration:
+                    break  # If no matching entry is found, break the loop
+
+                if entry.type == pygit2.GIT_OBJECT_TREE:
+                    # If the entry is a tree, update the current tree to this entry
+                    current_tree = repo[entry.id]
+                elif entry.type == pygit2.GIT_OBJECT_BLOB:
+                    # If the entry is a blob and it's the last part, return the entry
+                    if i == len(path_parts) - 1:
+                        return entry
+                    else:
+                        break  # If it's not the last part, break the loop
                 else:
-                    return None
-            else:
-                return None
+                    break  # If the entry is neither a tree nor a blob, break the loop
 
-    return found_entry
+    # Return None if no valid file is found
+    return None
```

Comment:
I like this `break` return None pattern better than the previous return None in else clause

Reply:
Thanks I'll take a look here and see about improving the previous None return.

Reply:
Ah sorry, I think I misunderstood you to mean we should follow this pattern elsewhere in the function. Thanks for the kind mention!